[유휘찬] Refactor: 로그인페이지 id, pw 로직 변경

chan-97 opened the pull request:
- 로그인, 메인페이지 리액트화
- 로그인페이지  assignment 조건에 맞는 버튼 활성화
- 메인페이지 댓글 기능 구현

>wecode PR message template 및 체크 리스트 입니다. 
아래 사항들을 전부 작성/체크 하시고 PR 해주세요!

## 수정 사항 간략한 한줄 요약
(여기에 수정 사항에 대한 간략한 한줄 요약/제목 작성해 주세요.)
- 로그인 버튼 활성화 기능 구현
		 
## 수정 사항들 자세한 내용
(여기에 수정 사항에 대한 자세한 내용을 작성해 주세요.)
- id '@' 포함, pw 5 글자 이상 입력 시 버튼 활성화

## 기타 질문 및 특이 사항
(궁금한 사항들, 하면서 느낀 점들 공유해주실 것이 있으면 자유롭게 작성해 주세요.)

## 체크 리스트 (아래 사항들이 전부 체크되어야만 merge가 됩니다!)
- [x] 필요한 test들을 완료하였고 기능이 제대로 실행되는지 확인 하였습니다.
- [x] Wecode의 코드 스타일 가이드에 맞추어 코드를 작성 하였습니다.
- [x] 제가 의도한 파일들과 수정 사항들만 커밋이 된 것을 확인 하였습니다.
- [x] 본 수정 사항들을 팀원들과 사전에 상의하였고 팀원들 모두 해당 PR에 대하여 알고 있습니다.

chan-97 committed:
[hwichanyu] ADD: 위스타그램 리액트화
- 로그인, 메인페이지 리액트화
- 로그인페이지  assignment 조건에 맞는 버튼 활성화
- 메인페이지 댓글 기능 구

joonsikyang commented on .vscode/settings.json:
```diff
@@ -0,0 +1,3 @@
+{
+    "liveServer.settings.port": 5502
+}
```
포트 번호를 세팅하신 이유가 있나요..?

chan-97 replied:
없습니다. 저게 왜 생겼는지는 모르겠지만 삭제하였습니다!

joonsikyang commented on src/Routes.js:
```diff
@@ -1,11 +1,11 @@
-import MainDongho from "pages/donghokim/Main/Main";
-import LoginDaseul from "pages/daseulsong/Login/Login";
-import MainDaseul from "pages/daseulsong/Main/Main";
+// import LoginYeji from "pages/yejikang/Login/Login";
+// import MainYeji from "pages/yejikang/Main/Main";
+// import LoginDongho from "pages/donghokim/Login/Login";
+// import MainDongho from "pages/donghokim/Main/Main";
+// import LoginDaseul from "pages/daseulsong/Login/Login";
+// import MainDaseul from "pages/daseulsong/Main/Main";
```
다른 분들 것 주석 처리하고 merge되면 안 됩니다. 주석 해제해주세요. 아래 Route 컴포넌트도 마찬가지입니다.

chan-97 replied:
수정햐였습니다

joonsikyang commented on src/pages/hwichanyu/Login/Login.js:
```diff
@@ -0,0 +1,78 @@
+  }
+
+  getValueAndGoToMain = () => {
+    console.log(this.state.id, this.state.pw);
```
console.log는 push 하기 전에 삭제 부탁드립니다.

chan-97 replied:
로그인 페이지 - 로그인 버튼 눌렀을 때, 사용자가 입력한 id와 pw 콘솔에 나올 수 있게 해주세요.
라는 assignment 가 있는데 지울까요....?

joonsikyang commented on src/pages/hwichanyu/Login/Login.js:
```diff
@@ -0,0 +1,78 @@
+    };
+  }
+
+  getValueAndGoToMain = () => {
```
함수 이름과 내용이 다르네요! `getValue` 하는 함수는 아니므로 `goToMain`이 적절할 것 같습니다.

chan-97 replied:
콘솔에 id 와 pw 를 찍어 확인하게 했으므로 confirmValAndGoToMain 으로 수정하였습니다.

joonsikyang commented on src/pages/hwichanyu/Login/Login.js:
```diff
@@ -0,0 +1,78 @@
+              />
+            </div>
+            {console.log(this.state)}
+            <button
+              className={
+                this.state.id.includes("@") && this.state.pw.length >= 5
+                  ? "buttonActive"
+                  : "justButton"
+              }
+              onClick={this.getValueAndGoToMain}
+            >
```
Goood! className에 삼항연산자 정확히 적용해주셨네요 :)

chan-97 replied:
감사합니다!!🤗🤗

joonsikyang commented on src/pages/hwichanyu/Login/Login.js:
```diff
@@ -0,0 +1,78 @@
+
+export default withRouter(LoginHwichan);
+
+{
+  /* disabled={this.state.id ? "false" : "trues"} */
+}
```
삭제해주세요.

joonsikyang commented on src/pages/hwichanyu/Login/Login.scss:
```diff
@@ -0,0 +1,106 @@
+@import "../../../styles/common.scss";
+@import "../../../styles/reset.scss";
+
+* {
+  box-sizing: border-box;
+}
```
해당 내용은 reset.scss에 더 잘어울리네요. 옮겨주세요.

joonsikyang commented on src/pages/hwichanyu/Login/Login.scss:
```diff
@@ -0,0 +1,106 @@
+@import "../../../styles/common.scss";
+@import "../../../styles/reset.scss";
```
- reset.scss도 import 할 필요가 있나요?
- 기본적으로 reset.scss, common.scss 는 index.js에서 한 번만 import 합니다.
- 공통 변수를 .scss 파일에서 사용해야 하는 경우엔 common.scss에서 변수를 관리해주기 때문에 common.scss만 import 하면 될 것 같습니다.

joonsikyang commented on src/pages/hwichanyu/Login/Login.scss:
```diff
@@ -0,0 +1,106 @@
+.Login {
+  height: 100vh;
+  margin: 0;
+  background-color: $background-grey;
+  @include center;
```
sass 활용 너무 잘하셨네요 🤘

joonsikyang commented on src/pages/hwichanyu/Login/Login.scss:
```diff
@@ -0,0 +1,106 @@
+    padding: 10px 0px;
+    border: 1px solid $border-grey;
+    background-color: $white;
+    opacity: 1;
```
opacity:1은 default 아닌가요?

joonsikyang commented on src/pages/hwichanyu/Main/Feeds/Feeds.js:
```diff
@@ -0,0 +1,140 @@
+import React, { Component } from "react";
+import "./Feeds.scss";
```
- .scss 파일은 맨 마지막에 import 해주세요.
- 보통 패키지 > 컴포넌트 > 변수/이미지 > 스타일 파일 순서로 import 합니다. 신경써주세요!

joonsikyang commented on src/pages/hwichanyu/Main/Feeds/Feeds.js:
```diff
@@ -0,0 +1,140 @@
+  constructor() {
+    super();
+    this.state = {
+      value: "",
+      list: [],
```
어떤 value 인지, 어떤 list 인지 key 이름에서 드러날 수 있게 수정해주세요.

joonsikyang commented on src/pages/hwichanyu/Main/Feeds/Feeds.js:
```diff
@@ -0,0 +1,140 @@
+  }
+
+  getCommVal = (e) => {
+    console.log(this.state.value);
```
console.log는 삭제 해주세요

joonsikyang commented on src/pages/hwichanyu/Main/Feeds/Feeds.js:
```diff
@@ -0,0 +1,140 @@
+    });
+  };
+
+  addComm = () => {
+    const userInfo = {
+      userId: "guest_",
+      comment: this.state.value,
+    };
+    this.setState({
+      list: [...this.state.list, userInfo],
+    });
+  };
```
Perfect!

joonsikyang commented on src/pages/hwichanyu/Main/Feeds/Feeds.js:
```diff
@@ -0,0 +1,140 @@
+
+  render() {
+    return (
+      <div className="feedsFrame">
```
최상위 요소의 className은 컴포넌트명과 동일하게 해주세요.

joonsikyang commented on src/pages/hwichanyu/Main/Feeds/Feeds.js:
```diff
@@ -0,0 +1,140 @@
+              <div>Hello_world</div>
+              <span>어느별에서 왔니??~~</span>
+            </li>
+            {this.state.list.map((comment) => {
+              return <AddComment userData={comment} />;
+            })}
```
Goood🙌

joonsikyang commented on src/pages/hwichanyu/Main/Feeds/Feeds.scss:
```diff
@@ -0,0 +1,252 @@
+@import "../../../../styles/reset.scss";
+@import "../../../../styles/common.scss";
+
+* {
+  box-sizing: border-box;
+}
```
마찬가지입니다. 위 코멘트 확인해주세요!

joonsikyang commented on src/pages/hwichanyu/Main/Feeds/Feeds.scss:
```diff
@@ -0,0 +1,252 @@
+  box-sizing: border-box;
+}
+
+html,
+body {
+  width: 100%;
+  height: 100%;
+}
+
+body {
+  margin: 0;
+}
+
+main {
+  height: 200vh;
+  background: $background-grey;
+}
+
+section {
+  width: 935px;
+  height: 100%;
+  display: flex;
+  margin: 0 auto;
+  padding-top: 30px;
+}
```
- Feeds.scss에 있어도 될까요? 
- section 태그만 보더라도 이렇게 nesting이 제대로 이루어지지 않으면 다른 파일의 section태그에 모두 영향을 줄 수 있습니다. 수정해주세요.

joonsikyang commented on src/pages/hwichanyu/Main/Feeds/Feeds.scss:
```diff
@@ -0,0 +1,252 @@
+  margin: 8px;
+}
+
+#feedheart {
+  margin-left: 0;
+}
+
+#feedBookmark {
+  margin-right: 0;
+}
```
id 사용하셨던 것들은 전부 className으로 수정 부탁드립니다.

joonsikyang commented on src/pages/hwichanyu/Main/Main.js:
```diff
@@ -0,0 +1,22 @@
+class Mymain extends Component {
+  render() {
+    return (
+      <div className="Main">
+        {/* <Nav></Nav> */}
+        <main>
+          <section className="mainFrame">
+            <Feeds></Feeds>
+            <Mainright></Mainright>
+          </section>
+        </main>
+      </div>
```
Main 컴포넌트인데 `<main>`태그를 최상위 요소로 하면 훨씬 좋겠네요!

joonsikyang commented on src/pages/hwichanyu/Main/Main.js:
```diff
@@ -0,0 +1,22 @@
+import Mainright from "./Mainright/Mainright";
+import "./Main.scss";
+
+class Mymain extends Component {
```
파일명과 컴포넌트명 동일하게 수정해주세요.

chan-97 replied:
수정 완료하였습니다

joonsikyang commented on src/pages/hwichanyu/Main/Mainright/Mainright.js:
```diff
@@ -0,0 +1,153 @@
+class Mainright extends Component {
+  render() {
+    return (
+      <div className="mainRight">
```
최상위 요소 className 컴포넌트명과 동일하게 잘 작성해주셨네요. 이것도 최상위 요소임을 나타내기 위해 대문자로 시작하게 수정해주세요.

chan-97 replied:
수정 완료 하였습니다

joonsikyang commented on src/pages/hwichanyu/Main/Mainright/Mainright.js:
```diff
@@ -0,0 +1,153 @@
+        </div>
+        <div className="about">
+          <ul>
+            <li>소개</li>
+            <span>·</span>
+            <li>도움말</li>
+            <span>·</span>
+            <li>홍보 센터</li>
+            <span>·</span>
+            <li>API</li>
+            <span>·</span>
+            <li>채용 정보</li>
+            <span>·</span>
+            <li>개인정보처리방침</li>
+            <span>·</span>
```
중간 중간 span 태그 안에 점을 포함시키기 보다는 css `:after` 기능으로 구현해보시기 바랍니다.

chan-97 replied:
가르쳐주셔서 감사합니다. 적용 완료하였습니다.

joonsikyang commented on src/pages/hwichanyu/Main/Mainright/Mainright.scss:
```diff
@@ -0,0 +1,171 @@
+// @import "../../../../styles/reset.scss";
```
필요 없으면 삭제해주세요!

chan-97 replied:
삭제하였습니다

joonsikyang commented on src/pages/hwichanyu/Main/Mainright/Mainright.scss:
```diff
@@ -0,0 +1,171 @@
+  margin-top: 17px;
+}
+
+@media only screen and (max-width: 1400px) {
+  .mainRight {
+    display: none;
+  }
+
+  .mainFrame {
+    display: flex;
+    justify-content: center;
+  }
+}
+
+@media only screen and (max-width: 950px) {
+  .mainFrame {
+    width: 700px;
+  }
+}
+
+@media only screen and (max-width: 700px) {
+  .feedsTop {
+    display: none;
+  }
+}
```
media query 💪

chan-97 replied:
😋

joonsikyang reviewed (commented):
리뷰 내용 확인 부탁드립니다 🙇🏻‍♂️

chan-97 committed:
[유휘찬] Modify : 로그인, 메인페이지 리뷰내용 적용
- <메인>
- 유저 아이디와 댓글 사이 공백 추가
- 하트버튼 클릭시 좋아요 수 증가
- 댓글 작성시 콘솔창에 뜨는 값 삭제
- 댓글기능 로직 수정
- 댓글 1글자 이상 작성시 게시버튼 색상 활성화
- 엔터버튼으로 댓글 입력기능 추가
- 댓글 추가시 기존 input value 삭제
- 댓글창에 입력한 값이 없으면 댓글이 추가되지 않게 기능 적용

dipsiiiiiiiiii commented on src/pages/hwichanyu/Login/Login.js:
```diff
@@ -0,0 +1,87 @@
+    e.preventDefault();
+    if (this.state.id === "wecode00@gmail.com" && this.state.pw === "777777") {
+      this.props.history.push("/main-hwichan");
+      console.log("id : " + this.state.id, "/ pw : " + this.state.pw);
```
테스트 끝난 콘솔은 지워주세요 :)

chan-97 replied:
수정했습니다~~😋

dipsiiiiiiiiii reviewed (commented):
기존 리뷰들 다 반영해주세요!

Kangyejiii commented on src/pages/hwichanyu/Main/Mainright/Mainright.scss:
```diff
@@ -136,6 +135,15 @@
   font-weight: 400;
   color: rgba(199, 199, 199, 1);
+
+  li::after {
+    content: "·";
+    margin: 0 2.5px;
+  }
```
📌  배워갑니다😊

chan-97 committed:
Refactor: 로그인페이지 id, pw 로직 변경
- input 데이터를 관리하는 메서드를 2개에서 1개로 수정
